SpinCore_pp/ppg/echo.py

```diff
@@ -19,24 +19,16 @@
 
 # {{{spin echo ppg
 def run_spin_echo(
-    nScans,
+    settings,
     indirect_idx,
     indirect_len,
-    adcOffset,
-    carrierFreq_MHz,
     nPoints,
-    nEchoes,
-    p90_us,
-    repetition_us,
-    tau_us,
-    SW_kHz,
+    plen,
     indirect_fields=None,
     ph1_cyc=r_[0, 1, 2, 3],
     ph2_cyc=r_[0],
     ret_data=None,
-    deadtime_us=10.0,
-    deblank_us=1.0,
-    amplitude=1.0,
+    plen_as_beta=True,
 ):
```

Comment:
if you do this, you're going to have to change *everything* that uses `run_spin_echo`, which are the following:
```
./SpinCore_pp/run_nutation_CPMG.py
./examples/run_CPMG.py
./examples/run_FS.py
./examples/run_largecyc_CPMG.py
./examples/run_var_tau.py
./examples/run_nutation.py
./examples/combined_ODNP.py
./examples/run_nutation_amplitude.py
./examples/run_Hahn_echo_mw.py
./examples/run_generic_echo.py
./examples/run_Hahn_echo.py
./examples/run_field_dep_justMw.py
./examples/run_largecyc_generic_echo.py
./Instruments/nmr_signal_gui.py
```
Is this the best way to go?

Since the generic sequence just takes raw pulse lengths, maybe it would be easier to go back to raw pulse lengths for `run_spin_echo` as well?

There's also a `run_IR` that would require a similar change.

I get why you're doing this,  but keep in mind that it will require changing a lot.